compiler/commands.nim

```diff
@@ -126,7 +126,15 @@ proc processOnOffSwitch(op: TOptions, arg: string, pass: TCmdLinePass,
   case whichKeyword(arg)
   of wOn: gOptions = gOptions + op
   of wOff: gOptions = gOptions - op
-  else: localError(info, errOnOrOffExpectedButXFound, arg)
+  else:
+    if arg == "list" and optHints in op:
```

Comment:
I could call it `--listHints`, but I didn't want to clutter command line options, so I preferred `--hints:list` and the same for warnings. That's why there are these special cases here.